src/fios_gui.cpp

```diff
@@ -666,8 +678,20 @@ struct SaveLoadWindow : public Window {
 			}
 		} else if (this->IsWidgetLowered(WID_SL_SAVE_GAME)) { // Save button clicked
 			if (this->abstract_filetype == FT_SAVEGAME || this->abstract_filetype == FT_SCENARIO) {
-				_switch_mode = SM_SAVE_GAME;
-				FiosMakeSavegameName(_file_to_saveload.name, this->filename_editbox.text.buf, lastof(_file_to_saveload.name));
+				if (_load_check_data.settings.game_creation.generation_unique_id != 0 && /* Don't warn if the save has no id (old save) */
+											_load_check_data.settings.game_creation.generation_unique_id != _settings_game.game_creation.generation_unique_id) {
+					/* The save has a different id to the current game */
+					/* Show a caption box asking whether the user is sure to overwrite the save */
+					ShowQuery(
+						STR_SAVEGAME_UNMATCHING_ID_CAPTION,
+						STR_SAVEGAME_UNMATCHING_ID_CONFIRMATION_TEXT,
+						this,
+						SaveGameConfirmationCallback
+										);
```

Comment:
Ending parenthesis should be on the previous line, no space before.
I think this entire call should just be a single line, however, no need to break it up.